src/utils/convert.js

```diff
@@ -825,278 +194,257 @@ const renderDefaultInputData = definition => {
+  const content = await renderStepTest(stepType, definition, appDefinition);
+  await createFile(content, filename, newAppDir);
+};
 
-  // Remove any 'use strict'; or "use strict"; since we add that automatically
-  templateContext.CODE = templateContext.CODE.replace(
-    "'use strict';\n",
-    ''
-  ).replace('"use strict";\n', '');
+const writeAuth = async (appDefinition, newAppDir) => {
+  const content = await renderAuth(appDefinition, appDefinition);
+  await createFile(content, 'authentication.js', newAppDir);
+};
 
-  const templateFile = path.join(TEMPLATE_DIR, '/scripting.template.js');
-  return renderTemplate(templateFile, templateContext, false);
+const writePackageJson = async (legacyApp, appDefinition, newAppDir) => {
+  const content = await renderPackageJson(legacyApp, appDefinition);
+  await createFile(content, 'package.json', newAppDir);
 };
 
-const writeScripting = (legacyApp, newAppDir) => {
-  return renderScripting(legacyApp).then(content => {
-    if (content) {
-      return createFile(content, 'scripting.js', newAppDir);
-    }
+const writeHydrators = async (appDefinition, newAppDir) => {
+  const content = await renderHydrators(appDefinition);
+  await createFile(content, 'hydrators.js', newAppDir);
+};
 
-    return null;
-  });
+const writeScripting = async (appDefinition, newAppDir) => {
+  await createFile(
+    appDefinition.legacy.scriptingSource,
+    'scripting.js',
+    newAppDir
+  );
 };
 
-const renderEnvironment = definition => {
-  const authFields = getAuthFieldKeys(definition);
-  const lines = _.map(authFields, key => {
-    const upperKey = _.snakeCase(key).toUpperCase();
-    return `${upperKey}=YOUR_${upperKey}`;
-  });
-  return lines.join('\n');
+const writeIndex = async (appDefinition, newAppDir) => {
+  const content = await renderIndex(appDefinition);
+  await createFile(content, 'index.js', newAppDir);
 };
 
-const writeEnvironment = (legacyApp, newAppDir) => {
-  const content = renderEnvironment(legacyApp);
-  if (!content) {
-    return Promise.resolve(null);
-  }
-  return createFile(content, '.env', newAppDir);
+const writeEnvironment = async (appDefinition, newAppDir) => {
+  const content = renderEnvironment(appDefinition);
+  await createFile(content, '.env', newAppDir);
 };
 
-const writeGitIgnore = newAppDir => {
+const writeGitIgnore = async newAppDir => {
   const srcPath = path.join(TEMPLATE_DIR, '/gitignore');
   const destPath = path.join(newAppDir, '/.gitignore');
-  return copyFile(srcPath, destPath).then(() => {
-    startSpinner('Writing .gitignore');
-    endSpinner();
-  });
+  await copyFile(srcPath, destPath);
+  startSpinner('Writing .gitignore');
+  endSpinner();
 };
 
-const convertApp = (legacyApp, newAppDir) => {
+const writeZapierAppRc = async newAppDir => {
+  const content = JSON.stringify({
+    includeInBuild: ['scripting.js']
+  });
+  await createFile(content, '.zapierapprc', newAppDir);
+  startSpinner('Writing .zapierapprc');
```

Comment:
presumably we'd want to start the spinner before beginning the async action?

Reply:
We're only using it to print messages and not really taking advantage of spinner here. Each output file is generated [concurrently with Promise.all](https://github.com/zapier/zapier-platform-cli/blob/4830272f39329485fc6a00ecb26aa3f691e6308b/src/utils/convert.js#L445). To properly use spinner to show progress, we need to create a spinner instance for each promise, which is not how we do it in `src/utils/display.js`. Since conversion is very fast, spinner doesn't help that much, I wouldn't bother to create multiple spinner instances just to properly show progress.